find.go

```diff
@@ -233,7 +233,13 @@ func (s *server) doFind(ctx context.Context, method, source string, req *url.URL
 			stats.WithMeasurements(metrics.FindLoad.M(1)))
 	}()
 
-	sg := &scatterGather[Backend, *model.FindResponse]{
+	// sgResponse is a struct that exists to capture the backend that the response has been received from
+	type sgResponse struct {
+		r *model.FindResponse
+		b Backend
+	}
+
+	sg := &scatterGather[Backend, sgResponse]{
```

Comment:
Since `Backend` is already an interface, it seems unnecessary to type `scatterGather` on `Backend`. Simpler to do:
```go
sg := &scatterGather[sgResponse]{ ... }
if err := sg.scatter(ctx, func(cctx context.Context, b Backend) (*sgResponse, error) { ... }
```
Since `Backend` is an interface, it does not need a generic type to pass it into the function.  This can be done is a separate PR, if it makes sense to do.